clio-server/src/main/scala/org/broadinstitute/clio/server/service/PersistenceService.scala

```diff
@@ -36,14 +38,14 @@ class PersistenceService private[server] (
     metadata: M,
     documentMapper: ElasticsearchDocumentMapper[Key, M],
     index: ElasticsearchIndex[_]
-  )(implicit ec: ExecutionContext): Future[UpsertId] = {
+  )(implicit ec: ExecutionContext): Source[UpsertId, NotUsed] = {
```

Comment:
Looking at this more, I think it'd be OK for this to return `Future` and have the calling code in `IndexService` do the `Source.fromFuture` wrapping. There's nothing inherently stream-y about this method so having it return `Source` can be misleading.